service/api.go

```diff
@@ -263,6 +264,12 @@ func (as *APIServer) validateProjectConfig(w http.ResponseWriter, r *http.Reques
 
 	if input.Quiet {
 		errs = errs.AtLevel(validator.Error)
+	} else if projectRef != nil || !isConfigDefined {
```

Comment:
Is this supposed to be `projectRef == nil`?

Reply:
Also, should this skip alias validation when there's no project config? I thought that the project config was optional.

Reply:
you are so right

Reply:
`ConstructMergedAliasesByPrecedence` is meant to return a list of aliases regardless of whether the project config is defined so yes, we'd want to do alias validation regardless